osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -62,4 +73,26 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
 
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+		for (VirtualSystem vs : da.getVirtualSystems()) {
+			if (!vs.getServiceFunctionChains().isEmpty()) {
```

Comment:
move this to a private method named validateVSReferenceToSfcAndSecurityGroup. In that method you can return immediately if the getSFCs is empty. This will reduce if-nesting and improve readability.

Reply:
Thank you for your review comments .I moved it into new suggested method.